Code/Tools/RemoteConsole/Core/RemoteConsoleCore.cpp

```diff
@@ -247,43 +266,42 @@ void SRemoteServer::Run()
 /////////////////////////////////////////////////////////////////////////////////////////////
 void SRemoteServer::AddEvent(IRemoteEvent* pEvent)
 {
-    m_lock.Lock();
+    AZStd::lock_guard<AZStd::recursive_mutex> lock(m_mutex);
     for (TClients::iterator it = m_clients.begin(); it != m_clients.end(); ++it)
     {
         it->pEvents->push_back(pEvent->Clone());
     }
-    m_lock.Unlock();
     delete pEvent;
 }
 
 /////////////////////////////////////////////////////////////////////////////////////////////
 void SRemoteServer::GetEvents(TEventBuffer& buffer)
 {
-    m_lock.Lock();
+    AZStd::lock_guard<AZStd::recursive_mutex> lock(m_mutex);
```

Comment:
repeat: Can be replaced with

```suggestion
    AZStd::scoped_lock lock(m_mutex);
```